src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -568,6 +568,7 @@ MulticopterPositionControl::parameters_update(bool force)
 		param_get(_params_handles.thr_min, &_params.thr_min);
 		param_get(_params_handles.thr_max, &_params.thr_max);
 		param_get(_params_handles.thr_hover, &_params.thr_hover);
+		_params.thr_hover = math::constrain(_params.thr_hover, _params.thr_min, _params.thr_max);
 		param_get(_params_handles.alt_ctl_dz, &_params.alt_ctl_dz);
```

Comment:
I totally support migration to the matrix library. That's why I converted the dot product calculation to the matrix library convention such that it will be compatible to the migration. Nevertheless I think min, max, constrain ... is not purpose of the matrix library.